packages/flutter/lib/src/rendering/proxy_sliver.dart

```diff
@@ -370,3 +372,26 @@ class RenderSliverOffstage extends RenderProxySliver {
     ];
   }
 }
+
+/// Makes its sliver child partially transparent, driven from an [Animation].
+///
+/// This is a variant of [RenderSliverOpacity] that uses an [Animation<double>]
+/// rather than a [double] to control the opacity.
+class RenderSliverAnimatedOpacity extends RenderProxySliver with RenderAnimatedOpacityMixin<RenderSliver>{
+  /// Creates a partially transparent render object.
+  ///
+  /// The [opacity] argument must not be null.
+  RenderSliverAnimatedOpacity({
+    @required Animation<double> opacity,
+    bool alwaysIncludeSemantics = false,
+    RenderSliver sliver,
+  }) : assert(opacity != null),
+       assert(alwaysIncludeSemantics != null) {
+    this.opacity = opacity;
+    this.alwaysIncludeSemantics = alwaysIncludeSemantics;
+    child = sliver;
+  }
+
+  @override
+  void paintWithOpacity(PaintingContext context, Offset offset) => context.paintChild(child, offset);
```

Comment:
This seems to be the same as the non sliver one, and they are also the same as super.paint.
Is this override necessary?

Reply:
I haven't found a way to specify the _right_ super.paint for the non-sliver class. The non-sliver class has another mixin (`RenderProxyBoxMixin`) via `RenderProxyBox` in addition to this proposed `RenderAnimatedOpacityMixin`. When using super.paint, it will use the later method instead of the former - which is the one it should call. 🤔 

For the sliver one, super.paint is a no-op. It uses context.paintChild.

Reply:
I am not sure if I understand. If you use super.paint in RenderAnimatedOpacityMixin.paint it should call RenderProxyBoxMixin.paint for non silver case, which is the paint we want, right? I wrote a demo code to demonstrate this

```dart
mixin amix on a {
  void test() {
    print('amix');
  }
}

mixin bmix on b {
  @override
  void test() {
    print('bmix');
    // This should print amix;
    super.test();
  }
}
abstract class a {
  a();
  void test();
}

class b extends a with amix {
}

class c extends b with bmix {

}

void main() {
  final c instance = c();
  // this should print bmix and amix
  instance.test();
}
```